dashboard/test/controllers/activities_controller_test.rb

```diff
@@ -893,7 +893,7 @@ def expect_controller_logs_milestone_regexp(regexp)
     )
     script = Script.add_script({name: 'Milestone Script'}, script_dsl[0][:lesson_groups], script_dsl[0][:lessons])
 
-    last_level_in_first_stage = script.lessons.first.script_levels.last
+    last_level_in_first_stage = script.lesson_group.first.lessons.first.script_levels.last
```

Comment:
one idea would be to update the `lessons` association to go through the lesson_groups association, so that you could leave this as `lessons.first`